superset-frontend/src/components/ListView/TableCollection.tsx

```diff
@@ -87,72 +200,74 @@ export default function TableCollection({
   headerGroups,
   rows,
   loading,
-}: Props) {
+}: TableCollectionProps) {
   return (
-    <Table {...getTableProps()} className="table table-hover">
-      <thead>
-        {headerGroups.map(headerGroup => (
-          <tr {...headerGroup.getHeaderGroupProps()}>
-            {headerGroup.headers.map(column => {
-              let sortIcon = <Icon name="sort" />;
-              if (column.isSorted && column.isSortedDesc) {
-                sortIcon = <Icon name="sort-desc" />;
-              } else if (column.isSorted && !column.isSortedDesc) {
-                sortIcon = <Icon name="sort-asc" />;
-              }
-              return column.hidden ? null : (
-                <th
-                  {...column.getHeaderProps(
-                    column.canSort ? column.getSortByToggleProps() : {},
-                  )}
-                  data-test="sort-header"
-                  className={cx({
-                    [column.size || '']: column.size,
-                  })}
-                >
-                  <span>
-                    {column.render('Header')}
-                    {column.canSort && sortIcon}
-                  </span>
-                </th>
-              );
-            })}
-          </tr>
-        ))}
-      </thead>
-      <tbody {...getTableBodyProps()}>
-        {rows.map(row => {
-          prepareRow(row);
-          return (
-            <tr
-              {...row.getRowProps()}
-              className={cx('table-row', {
-                'table-row-selected': row.isSelected,
-              })}
-            >
-              {row.cells.map(cell => {
-                if (cell.column.hidden) return null;
-
-                const columnCellProps = cell.column.cellProps || {};
-                return (
-                  <td
-                    className={cx('table-cell', {
-                      'table-cell-loader': loading,
-                      [cell.column.size || '']: cell.column.size,
+    <TableContainer>
+      <Table {...getTableProps()} className="table table-hover">
+        <thead>
+          {headerGroups.map(headerGroup => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => {
+                let sortIcon = <Icon name="sort" />;
```

Comment:
Not sure if this is easier or harder to parse ¯\\\_(ツ)_/¯ :
`<Icon name={column.isSorted ? (column.isSortedDesc ? "sort-desc" : "sort-desc") : "sort" } />`

Also, you _could_ just set the value of `sortIconName` and then use it inline later in rendering, i.e. `{column.canSort && <Icon name={sortIconName} />`

Reply:
These are just indentation changes, so sorry if I'm off target ;)

Reply:
i'm not sure we're allowed nested ternaries per our eslint rules. I've already caused a bug trying to "simplify" this logic once, I'd like to avoid touching this logic again unless necessary.